workshop-resources/keyboard-navigation/event-reviews/2022-11-21-jupyteraccessibility.md

```diff
@@ -0,0 +1,222 @@
+2. Once focus is on the skip link, press Space or Enter to activate it.  
+
+Is there a skip link?
+Result:
+
+Is the skip link prompt visible?
+Result:
+
+What does the skip link prompt skip?
+Result:
+
+Where does the skip link prompt move user focus to?
+Result:
+
+Does the skip link behavior meet reviewer expectations?
+Result:
+
+Other notes or recommendations:
+
+### Interactive Areas
+
+All buttons, links, form fields, or similar interactive areas can be both accessed and activated using only the keyboard. ([WCAG 2.2 Keyboard](https://www.w3.org/TR/WCAG22/#keyboard))
+
+1. Navigate via the Tab key to interactive areas. 
+2. Activate interactive areas using the Enter or Spacebar key.
+
+| Interactive area | Able to navigate to the area (yes/no/not applicable) | Able to input information (yes/no/not applicable) | Able to activate the area (yes/no/not applicable) |
+|---|---|---|---|
+| Button 1| a | b | c |
+| Button 2| d | e | f |
+| Button 3 | g | h | i |
+| Link 1 | j | k | l |
+| Link 2 | m | n | o |
+| Form field 1 | p | q | r |
+| Add table rows as needed! |  |  |  |
+| |  |  |  |
+
+Other notes by area:
+
+### Focus
+
+There is a visible focus state for all content. It appears by default when navigating via keyboard. ([WCAG 2.2 Focus visible](https://www.w3.org/TR/WCAG22/#focus-visible))
+
+1. From the top of the page, press the Tab key repeatedly until reviewer reaches the end of content. 
+2. If end of content cannot be reached, the reviewer may report on what they could interact with.
+3. If the reviewer is comfortable, they may also use the browser inspector to select an area and toggle focus per area.
+
+Is there visible focus styling? If so, please briefly describe it.
+Result:
+
+Is the focus styling consistent throughout?
+Result:
+
+Are there any areas without visible focus styling?
+Result: 
+
+Was the reviewer ever unclear about where their keyboard focus was during the review? If so, please note where or why.
+Result:
+
+Other notes or recommendations:
+
+### Mixed input
+
+It is possible to complete tasks while switching between input mechanisms, for example using a keyboard and mouse and touch screen. ([WCAG 2.2 Concurrent input mechanisms](https://www.w3.org/TR/WCAG22/#concurrent-input-mechanisms))
+
+1. Locate an interactive area.
+2. Complete the interaction using keyboard controls.
+3. Complete the interaction using the mouse.
+4. Complete the interaction using touch controls.
+5. If the task has multiple steps, the reviewer can try using a different interaction for each step.
+6. Repeat as needed across interactive areas.
+
+#### Task name + region
+
+Repeat this section as needed for each task or region the reviewer evaluates.
+
+Does the task have multiple steps? If yes, please list them.
+Result:
+
+Can the task be completed with only keyboard controls?
+Result:
+
+Can the task be completed with only mouse controls?
+Result:
+
+Can the task be completed with only touch screen contorls?
+Result:
+
```

Comment:
```suggestion
#### Page drop down menus (ie. GitHub and Download icons)

Does the task have multiple steps? If yes, please list them.
Result: Tab to focus the page menu (GitHub and Download icons), reach focus for a menu with a drop down available, press space or enter, drop down menu appears, navigate drop down menu, select drop down item.

Can the task be completed with only keyboard controls?
Result: No. The menus can be tabbed to, but they cannot be activated to open the drop down with the keyboard. They can be activated and navigated with a mouse. Even if the menus are opened with a mouse, the user cannot switch to navigate them with a keyboard.

I was also surprised because the left and right arrow keys—typically used for fine-grain navigation through things like drop down menu items—immediately move me to the next page in the documentation. I am guessing this is because Jupyter Book is modeled like a book where turning pages to continue reading is high priority. The fact that it overrides keyboard navigation, concurrent with other navigation methods or not, is a problem.

Can the task be completed with only mouse controls?
Result: Yes.

Can the task be completed with only touch screen controls?
Result: n/a

```